### ISSUE TITLE
[BUG]d435i works, but not completely on NX devkit

### ISSUE THREAD

**Issue by AndreV84** [jetsonhacks/installRealSenseSDK#9]
**Describe the issue**
Please describe the issue
I tried both building from sources [ in which case it wouldn't run rs-distance]
and apt installation [ in which case some of examples worked, but not reasense viewer]
**What version of L4T/JetPack**
L4T/JetPack version: 4.4

**Which Intel RealSense camera(s)**
RealSense camera(s):d435i

**To Reproduce**
For example, what command line did you run?
for source code build installation rs-distance won't work
for apt installation other examples won't run, but rs-distance works

**Expected behavior**
realsense-viewer is expected to work both in case with installation from apt repository and from sources; Moreover, other graphical examples would be anticipated to work also
**Additional context**
Have you had a chance to try D400x series on NX?

**Comment by patrickpoirier51** [comment]
Having same issue, standing by for tests

**Comment by AndreV84** [comment]
actually after upgrading the firmware to the latest realsense-viewer seems to work better;
However, there are still some complications with ros zed wrapper docker container standalone node

**Comment by patrickpoirier51** [comment]
There is an issue with the UVC driver and the Kernel, even with latest RealSense-viewer you need to hiave to reset between modes

**Comment by patrickpoirier51** [comment]
@jetsonhacks  are you working on an update for the XAVIER NX Kernel patch so it can work with Realsense UVC , as the apt get librealsense-dkms does not work ?

Would the old XAVIER method of building kernel could be used?

**Comment by AndreV84** [comment]
@patrickpoirier51 
did you try upgrade the firmware of the camera? after the upgrade there shouldn't be no issues, in my opinion

**Comment by patrickpoirier51** [comment]
@AndreV84 
Yes I did, Not having the kernel patched for UVC makes it impossible to properly work with the camera

The RealSense-viewer do reset the hardware in order to get the camera connected, but that is impractical for other applications , namely python

**Comment by patrickpoirier51** [comment]
@jetsonhacks  are you working on an update for the XAVIER NX Kernel patch so it can work with Realsense UVC , as the apt get librealsense-dkms does not work ?

Would the old XAVIER method of building kernel could be used?

**Comment by patrickpoirier51** [comment]
@AndreV84  just to make sure, i have firmware release 05.12.03.00 on camera

**Comment by AndreV84** [comment]
5.12.5.0 is the latest, I believe https://www.intelrealsense.com/download/11232/?_ga=2.160502397.1537282655.1592577394-1689438527.1589854667

**Comment by patrickpoirier51** [comment]
@AndreV84  ok thanks, just updated (with my I7 based PC) and tried on Xavier NX
Here is a sequence that can make it crash
rs-pointcloud
ctrl-c and then call
rs-depth
and then ctrl-c and call rs-pointcloud again

The application crash -need to FORCE QUIT-  and DMESG shows:
attempt power cycle
[ 1556.606838] usb 2-3.4: Device not responding to setup address.
[ 1556.814884] usb 2-3.4: Device not responding to setup address.
[ 1557.021654] usb 2-3.4: device not accepting address 111, error -71
[ 1557.350923] usb 2-3.4: Device not responding to setup address.
[ 1557.558891] usb 2-3.4: Device not responding to setup address.
[ 1557.765633] usb 2-3.4: device not accepting address 112, error -71
[ 1557.766552] usb 2-3-port4: unable to enumerate USB device
[ 1558.242842] usb 2-3.4: Device not responding to setup address.
[ 1558.450833] usb 2-3.4: Device not responding to setup address.
[ 1558.657682] usb 2-3.4: device not accepting address 113, error -71
[ 1558.986832] usb 2-3.4: Device not responding to setup address.
[ 1559.194826] usb 2-3.4: Device not responding to setup address.
[ 1559.401649] usb 2-3.4: device not accepting address 114, error -71

Need to disconnect - reconnect to make it work again

**Comment by AndreV84** [comment]
@patrickpoirier51 did you update the firmware to 5.12.5 ?

**Comment by patrickpoirier51** [comment]
Yes, I did, _**just updated (with my I7 based PC) and tried on Xavier NX**_

**Comment by AndreV84** [comment]
sigh;
most of the  issues with restarting the camera application should have been gone by the upgrade, as far as I know; I can not test it at the moment; but will try to do so when I have access to the hardware

**Comment by AndreV84** [comment]
I used to upgrade the firmware from Jetson with
rs-fw-update

**Comment by patrickpoirier51** [comment]
I used the realsense-viewer , its quite the same

**Comment by patrickpoirier51** [comment]
rs-fw-update -l

connected devices:
1) Name: Intel RealSense D435I, serial number: 938422076009, update serial number: 926623051511, firmware version: 05.12.05.00, USB type: 3.2

**Comment by mzahana** [comment]
@patrickpoirier51 Nice to see you here. It's been long time!

I would like to use the D435i with Xavier NX, but in a ROS environment inside a docker container. Are there some instructions on how to that, or at least how to setup the Librealsense such that D435i works properly with NX?

Thanks.

**Comment by patrickpoirier51** [comment]
@mzahana  hello

Unfortunately I havent tried if Realsense dos behave correctly within docker on JetPack.
 Looking on google , we can find this one that may get you jumpstarted
https://broerkens.de/RACECARX/docs/software/jetson_docker

**Comment by mzahana** [comment]
@patrickpoirier51 thank you very much! I will give it a try and share my findings
